biz.aQute.bnd.annotation/src/aQute/bnd/annotation/service/ServiceCapability.java

```diff
@@ -0,0 +1,29 @@
+package aQute.bnd.annotation.service;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.osgi.annotation.bundle.Capability;
+import org.osgi.namespace.service.ServiceNamespace;
+import org.osgi.resource.Namespace;
+
+/**
+ * Adds a ProvideCapability for a service. This is useful to the resolver, when
+ * a service is registered manually. It can annotate a class or a package. If it
+ * annotates a class without specifying a value, the class will be named as the
+ * registered service. Uses constraints will be calculated automatically.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target({
+	ElementType.TYPE, ElementType.PACKAGE
+})
+@Capability(namespace = ServiceNamespace.SERVICE_NAMESPACE, //
+	attribute = {
+		ServiceNamespace.CAPABILITY_OBJECTCLASS_ATTRIBUTE + ":List<String>=\"${uniq;${#value}}\"", //
+		Namespace.CAPABILITY_USES_DIRECTIVE + ":=\"${uniq;${replace;${#value};(.*)\\.[^.]+;$1}}\""
+	})
+public @interface ServiceCapability {
+	Class<?>[] value() default Target.class;
```

Comment:
I feel this could be a little richer such as including `uses`, `attributes` which map into @Capability.*. We have this on `@ServiceProvider` https://github.com/bndtools/bnd/blob/master/biz.aQute.bnd.annotation/src/aQute/bnd/annotation/spi/ServiceProvider.java as an example.

Reply:
@rotty3000  Makes sense. I would try to build it in such a manner, that uses can be overwritten. Would you thing the following would work (and might be a useful improvement for ServiceProvider

`Namespace.CAPABILITY_USES_DIRECTIVE + ":=\"${if;${size;${#uses}};'${#uses}'; ${uniq;${replace;${#value};(.*)\\.[^.]+;$1}}}\"`

What do you think?

Reply:
It looks fine but you could just copy the exact code from the @ServiceProvider since the logic should be identical.

Reply:
ah hmm, perhaps that's enough for `uses`. But for attributes then we need a way to input and there you could copy existing example.